include/itkMultiScaleHessianEnhancementImageFilter.hxx

```diff
@@ -0,0 +1,248 @@
+  typename TOutputImage::Pointer outputImageTypePointer;
+
+  /* Process the first scale */
+  outputImageTypePointer = generateResponseAtScale((SigmaType)0);
+
+  /* Process the remaining sigma values */
+  for (SigmaStepsType scaleLevel = 1; scaleLevel < m_SigmaArray.GetSize(); ++scaleLevel)
+  {
+    /* Calculate next response value */
+    typename TOutputImage::Pointer tempResponseOutputImageTypePointer = generateResponseAtScale(scaleLevel);
+
+    /* Take absolute value maximum */
+    m_MaximumAbsoluteValueFilter->SetInput1(outputImageTypePointer);
+    m_MaximumAbsoluteValueFilter->SetInput2(tempResponseOutputImageTypePointer);
+    m_MaximumAbsoluteValueFilter->Update();
+
+    /* Save max and go to next sigma value */
+    outputImageTypePointer = m_MaximumAbsoluteValueFilter->GetOutput();
+  }
+
+  /* Graft output and we're done! */
+  this->GraftOutput(outputImageTypePointer);
+}
+
+template< typename TInputImage, typename TOutputImage >
+typename TOutputImage::Pointer
+MultiScaleHessianEnhancementImageFilter< TInputImage, TOutputImage >
+::generateResponseAtScale(SigmaStepsType scaleLevel)
+{
+  /* Get this sigma value */
+  SigmaType thisSigma = m_SigmaArray.GetElement(scaleLevel);
+
+  /* Process pipeline and return */
+  m_HessianFilter->SetSigma(thisSigma);
+  m_EigenToScalarImageFilter->Update();
+  return m_EigenToScalarImageFilter->GetOutput();
+}
+
+template< typename TInputImage, typename TOutputImage >
+typename MultiScaleHessianEnhancementImageFilter< TInputImage, TOutputImage >::SigmaArrayType
+MultiScaleHessianEnhancementImageFilter< TInputImage, TOutputImage >
+::GenerateSigmaArray(SigmaType SigmaMinimum, SigmaType SigmaMaximum, SigmaStepsType NumberOfSigmaSteps, SigmaStepMethodEnum SigmaStepMethod)
+{
+  /* Quick check to make sure value is correct */
+  if ( NumberOfSigmaSteps < 1 )
+  {
+    /*
+    * Since this is a static class, we have no "this" pointer so we cannot throw an exception. Instead
+    * we will return an empty sigma array.
+    */
+    // itkExceptionMacro(<< "NumberOfSigmaSteps (" << NumberOfSigmaSteps << ") must be one or greater");
+    SigmaArrayType sigmaArrayEmpty;
+    sigmaArrayEmpty.SetSize(0);
+    return sigmaArrayEmpty;
+  }
+
+  /* If we have min greater than max just swap */
+  if (SigmaMinimum > SigmaMaximum) {
+    std::swap(SigmaMinimum, SigmaMaximum);
+  }
+
+  /* If min equals max, just generate one step size */
+  if (SigmaMinimum == SigmaMaximum) {
+    NumberOfSigmaSteps = 1;
+  }
+
+  /* Create array and resize */
+  SigmaArrayType sigmaArray;
+  sigmaArray.SetSize(NumberOfSigmaSteps);
+
+  /* Populate that array */
+  SigmaType thisSigma;
+  for (SigmaStepsType scaleLevel = 0; scaleLevel < sigmaArray.GetSize(); ++scaleLevel)
+  {
+    switch ( SigmaStepMethod )
+    {
+    case Self::EquispacedSigmaSteps:
+      {
+      const RealType stepSize = std::max( 1e-10, ( SigmaMaximum - SigmaMinimum ) / ( NumberOfSigmaSteps - 1 ) );
```

Comment:
Division by 0 if NumberOfSigmaSteps is 1

Reply:
Good find. Strange that it passed the tests... See ce229e2d94ba40f90588298a8e0db649bf84e491